crates/polars-lazy/src/physical_plan/expressions/apply.rs

```diff
@@ -509,6 +509,12 @@ impl ApplyExpr {
                     let min = st.to_min()?;
                     let max = st.to_max()?;
 
+                    if ChunkCompare::equal(max, min).ok()?.all() {
```

Comment:
Sorry for the second iteration here. But using `AnyValue` comparison here will be faster and likely more readable.

Something like `max.get(0) == min.get(0)`